packages/web/app/components/Field/FileChooserField.jsx

```diff
@@ -25,10 +37,68 @@ const ChangeSelectionButton = styled.a`
   cursor: pointer;
 `;
 
+// Smaller string with ellipsis in the middle to show file extension
+const getSmallFileName = (value, maxLength) => {
+  const middlePoint = Math.floor(maxLength / 2);
+  const ellipsisOffset = 3;
+  const lastHalfIndex = value.length - middlePoint + ellipsisOffset;
+  return value.slice(0, middlePoint) + '...' + value.slice(lastHalfIndex, value.length);
+};
+
+const ValueSection = ({ value, smallDisplay, showFileDialog, onClear }) => {
+  if (smallDisplay) {
+    const maxLength = 50;
+    const needEllipsis = value.name.length > maxLength;
+    const smallName = needEllipsis ? getSmallFileName(value.name, maxLength) : value.name;
```

Comment:
TIL "ellipsis" refers to the whole cluster of three dots, I always assumed that was the plural ellipses!